Possible `checkCommit` panics under heavy traffic

#### System information

Geth version: `v0.5.2-stable`
OS & Version: Windows/Linux/OSX

#### Expected behaviour

DBFT should handle and broadcast block messages without panic.

#### Actual behaviour

When receiving huge amount of transactions (here we sent Blob transactions), dBFT reported a panic due to consensus verification failure.

#### Steps to reproduce the behaviour

1. Setup a multi-CN Neo X network;
2. Send a lot of transactions to the network, and cause `missing tx`.

#### Backtrace

````
INFO [03-05|08:20:05.910] Imported new potential chain segment     number=33583 hash=df6949..a3a0b5 blocks=1 txs=9  mgas=5.964  elapsed=119.133ms    mgasps=50.060  snapdiffs=4.11MiB   triediffs=31.30MiB  triedirty=165.22MiB
INFO [03-05|08:20:05.914] Chain head was updated                   number=33583 hash=df6949..a3a0b5 root=925de8..98b883 elapsed=4.1713ms
INFO [03-05|08:20:05.917] Starting work on payload                 id=0x030e281d30b103b6
INFO [03-05|08:20:05.930] Updated payload                          id=0x030e281d30b103b6 number=33584 hash=e9dc8c..41450c txs=0  withdrawals=0 gas=0          fees=0          root=9acb6e..19be04 elapsed=13.278ms
INFO [03-05|08:20:05.932] Commit new sealing work                  number=33584 sealhash=a9d590..b43526 txs=0  gas=0          elapsed=15.342ms
INFO [03-05|08:20:06.063] New block in the chain                   "dbft index"=33583 "chain index"=33583 hash=0xdf6949809a40e1c94a55ece8ce7d006391e6b2bbe78a803f71266182e7a3a0b5 "parent hash"=0x5ee281411a2b6385f45b1c2c4d7b1608149e464362848128c86bdbf6cb58ffd2 primary=4 coinbase=0x1212000000000000000000000000000000000003 "mix digest"=0xc666e49830bb59bc87d694c5eb2c049dd197dc6abf4fad8bf25193e81c12185a
INFO [03-05|08:20:06.063] Fetching latest sealing proposal         "desired number"=33584
INFO [03-05|08:20:06.063] Sealing proposal updated                 number=33584 sealhash=a9d590..b43526 "parent hash"=df6949..a3a0b5 txs=0
INFO [03-05|08:20:06.063] DKG task watcher                         currentHeight=33583 watchListLength=0
2026-03-05T08:20:06.064Z	INFO	initializing dbft	{"height": 33584, "view": 0, "index": 6, "role": "Backup"}
2026-03-05T08:20:06.064Z	DEBUG	reset timer	{"h": 33584, "v": 0, "delay": "9.573241204s"}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PrepareResponse", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
2026-03-05T08:20:06.064Z	DEBUG	received message	{"type": "Commit", "from": 3, "height": 33583, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:06.064Z	DEBUG	ignoring old height	{"height": 33583}
INFO [03-05|08:20:07.681] Looking for peers                        peercount=8 tried=0 static=0
2026-03-05T08:20:10.633Z	DEBUG	received message	{"type": "PrepareRequest", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.633Z	INFO	received PrepareRequest	{"validator": 5, "tx": 7}
2026-03-05T08:20:10.633Z	INFO	missing tx	{"count": 7}
WARN [03-05|08:20:10.670] Unexpected transaction delivery          peer=dbd55d5f73a86465f9cb78f61fb59f96abe3c8bbbb208fc4e092e8c15533ba62
2026-03-05T08:20:10.705Z	DEBUG	received message	{"type": "PrepareResponse", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.705Z	INFO	received PrepareResponse	{"validator": 1}
2026-03-05T08:20:10.705Z	DEBUG	check prepare: some transactions are missing	{"hashes": ["0x7cb566c7a4eadfef77da4b1edd7b7c49c9e1398988897107d2c2dfccbde0fa42","0xf781b09b707756c0ca3b8867b5ee60745ae9a28eccf896d4783ca080d5fa74b3","0xc4c60c5b491d5d5011059415ad594e89e5a1129e949e866996254127293c01a3","0x9013abc7b6d306269319fb56821cc66c85461e401b19e69eba3ba60c9b64fbae","0xcf800b3c4faa6ce28d7d0570217a8e27d417ece80050ad5fed99ca0dc4a0ab67"]}
2026-03-05T08:20:10.705Z	DEBUG	received message	{"type": "PrepareResponse", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.705Z	INFO	received PrepareResponse	{"validator": 4}
2026-03-05T08:20:10.705Z	DEBUG	check prepare: some transactions are missing	{"hashes": ["0x7cb566c7a4eadfef77da4b1edd7b7c49c9e1398988897107d2c2dfccbde0fa42","0xf781b09b707756c0ca3b8867b5ee60745ae9a28eccf896d4783ca080d5fa74b3","0xc4c60c5b491d5d5011059415ad594e89e5a1129e949e866996254127293c01a3","0x9013abc7b6d306269319fb56821cc66c85461e401b19e69eba3ba60c9b64fbae","0xcf800b3c4faa6ce28d7d0570217a8e27d417ece80050ad5fed99ca0dc4a0ab67"]}
2026-03-05T08:20:10.705Z	DEBUG	received message	{"type": "PrepareResponse", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.705Z	INFO	received PrepareResponse	{"validator": 0}
2026-03-05T08:20:10.705Z	DEBUG	check prepare: some transactions are missing	{"hashes": ["0xf781b09b707756c0ca3b8867b5ee60745ae9a28eccf896d4783ca080d5fa74b3"]}
WARN [03-05|08:20:10.721] proposed PreBlock verification failed    err="failed to validate body: transaction root hash mismatch (header value 05995411e08495cd2d3c134c7e575e8cb92dbe51b9381946dbb5e9474a8245ef, calculated 58ae8cbc57be6b835e8bcce318514f30aab15f0e01230d609c5f3df5a1c236ef)"
2026-03-05T08:20:10.721Z	WARN	proposed preBlock fails verification
2026-03-05T08:20:10.721Z	DEBUG	reset timer	{"h": 33584, "v": 0, "delay": "20s"}
2026-03-05T08:20:10.721Z	INFO	request change view	{"view": 0, "height": 33584, "reason": "TxInvalid", "new_view": 1, "nc": 0, "nf": 2}
2026-03-05T08:20:10.721Z	DEBUG	broadcasting message	{"type": "ChangeView", "height": 33584, "view": 0}
2026-03-05T08:20:10.730Z	DEBUG	received message	{"type": "PrepareResponse", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.730Z	DEBUG	ignoring PrepareResponse	{"dup": false, "sor": true, "viewChanging": true, "moreThanF": false}
2026-03-05T08:20:10.741Z	DEBUG	received message	{"type": "PreCommit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.742Z	INFO	received PreCommit	{"validator": 0}
2026-03-05T08:20:10.742Z	DEBUG	not enough PreCommits to process PreBlock	{"count": 1}
2026-03-05T08:20:10.742Z	DEBUG	received message	{"type": "PreCommit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.742Z	INFO	received PreCommit	{"validator": 2}
2026-03-05T08:20:10.742Z	DEBUG	not enough PreCommits to process PreBlock	{"count": 2}
2026-03-05T08:20:10.742Z	DEBUG	received message	{"type": "PreCommit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.742Z	INFO	received PreCommit	{"validator": 5}
2026-03-05T08:20:10.743Z	DEBUG	not enough PreCommits to process PreBlock	{"count": 3}
2026-03-05T08:20:10.743Z	DEBUG	received message	{"type": "PreCommit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.743Z	INFO	received PreCommit	{"validator": 4}
2026-03-05T08:20:10.743Z	DEBUG	not enough PreCommits to process PreBlock	{"count": 4}
2026-03-05T08:20:10.748Z	DEBUG	received message	{"type": "PrepareResponse", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.761Z	DEBUG	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": true, "moreThanF": true}
2026-03-05T08:20:10.761Z	DEBUG	received message	{"type": "PreCommit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.761Z	INFO	received PreCommit	{"validator": 1}
2026-03-05T08:20:10.761Z	INFO	processing PreBlock	{"height": 33584, "view": 0, "tx_count": 7, "preCommit_count": 5}
2026-03-05T08:20:10.761Z	DEBUG	can't send commit since self preCommit not yet sent
2026-03-05T08:20:10.761Z	DEBUG	received message	{"type": "PreCommit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.761Z	DEBUG	received message	{"type": "PreCommit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.761Z	DEBUG	received message	{"type": "PreCommit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.761Z	DEBUG	received message	{"type": "Commit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.763Z	INFO	received Commit	{"validator": 0}
WARN [03-05|08:20:10.764] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.764Z	DEBUG	received message	{"type": "Commit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.766Z	INFO	received Commit	{"validator": 4}
WARN [03-05|08:20:10.771] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.774Z	DEBUG	received message	{"type": "Commit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.775Z	INFO	received Commit	{"validator": 1}
WARN [03-05|08:20:10.775] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.775Z	DEBUG	received message	{"type": "Commit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	INFO	received Commit	{"validator": 5}
WARN [03-05|08:20:10.777] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "PreCommit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	INFO	received Commit	{"validator": 2}
WARN [03-05|08:20:10.777] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.777Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.787Z	DEBUG	received message	{"type": "Commit", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.788Z	DEBUG	received message	{"type": "Commit", "from": 4, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.788Z	DEBUG	received message	{"type": "PrepareResponse", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.788Z	DEBUG	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": true, "moreThanF": true}
2026-03-05T08:20:10.788Z	DEBUG	received message	{"type": "Commit", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.788Z	DEBUG	received message	{"type": "Commit", "from": 5, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.788Z	DEBUG	received message	{"type": "Commit", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	DEBUG	received message	{"type": "ChangeView", "from": 3, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	INFO	received ChangeView	{"validator": 3, "reason": "TxInvalid", "new view": 1}
2026-03-05T08:20:10.806Z	DEBUG	received message	{"type": "PreCommit", "from": 3, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	INFO	received PreCommit	{"validator": 3}
2026-03-05T08:20:10.806Z	DEBUG	can't send commit since self preCommit not yet sent
2026-03-05T08:20:10.806Z	DEBUG	received message	{"type": "RecoveryMessage", "from": 3, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	DEBUG	recovery message received	{"dump": {"ValidBlockStart":0,"ValidBlockEnd":33584,"Sender":"0x910ad1641b7125eff746accdca1f11148b22f472","Data":"+QfIQYKDMAOA+Qe/+QFe+ESAuEFV+j+O53qVX0n8kc7rBcAyPbcTpzxAUHH+/qogPw09THHAVUrc0NmD5YEicstTCLuo0yp4hHn3iIO78kOU9iGMAfhEAbhBXs4HXRn3rv5SQh7qKFi/AYRhVQ9/ZM77/MTVzVEf1EUP76vyf8ndCpqaLWJqaldnuWHXbwvnHBMQVKeQp4DiDgH4RAK4QdcMsfe/Wro2g9IJ3YWWHWckKodNWdpnanHkJuwYTD3cSOtK/jV786vL/SiA2tXdi6cFSnt0Zuv6hXdhuPlY8fgA+EQEuEEuMF5jhr69+wosoGPBzyeLawwfWbeaunlAbnmR/KX982b3I+WH4LMdEkj+G1BqK2H5jfN5SZSU2Se5A9KqBbobAfhEBbhBAt7pcBzgh0fOlr1eASsxkAI0Xm7oIqv22GrhwdAFPu0oslnkkj3Yh2Tk3pRrov7jkidYs/UesmlHDsF0txoi6gHAwKDCjMwoJjmP8CNafbN0BK0nJ3asH7CeXQ6NI78JX6DtVPkEVSCAgID5BE75Ao2g32lJgJpA4clKVezozn0AY5HmsrvnioA/cSZhguejoLWgHcxN6N7HXXqrhbVntszUGtMSRRuUinQT8KFC/UDUk0eUEhIAAAAAAAAAAAAAAAAAAAAAAAOgy4hDoHomM96fFL0HQnveJtk5GR7A6SO9Du5pP4UQv5+gBZlUEeCElc0tPBNMfldejLktvlG5OBlG27XpR0qCRe+gtos7QLEXmBAUUj99E8VIOFB7FI17dDmcpb3r0LV38IC5AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAoKDMIQCJVEAgwI+OIRpqTy2ogIBPki8jy61AiZ7Hq5r3tLs4xopgpzflK+0TzdohwJkPkOgxmbkmDC7WbyH1pTF6ywEndGX3Gq/T62L8lGT6BwSGFqIAAAAAAAAAAWFBKgXyACgVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCGDDgAAgzQAAKBW6B8XG8xVpv+DReaSwPhuW0jgG5lsrcABYi+142O0IaDjsMRCmPwcFJr79MiZb7kkJ65B5GSbk0yklZkbeFK4VfjnoHy1Zsek6t/vd9pLHt17fEnJ4TmJiIlxB9LC38y94PpCoPeBsJtwd1bAyjuIZ7XuYHRa6aKOzPiW1Hg8oIDV+nSzoMTGDFtJHV1QEQWUFa1ZTonloRKelJ6GaZYlQScpPAGjoDR2FBZvmp0Spq9x1hV2zrEPD7OLw6Mx4/C3drruc8mboJATq8e20wYmkxn7VoIcxmyFRh5AGxnmnro7pgybZPuuoECwWfTxxU8tod6ULZ9pEOpOk1p7P+17pPRtQbd+HFUIoM+ACzxPqmzijX0FcCF6jifUF+zoAFCtX+2Zyg3EoKtnoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAuLICAT5IvI8utQImex6ua97S7OMaKYKc35SvtE83aIcCZD5DholotmwF98EmYhMcOaRFXnXK3ZKL5ivA7ksnYlPSlI4/RUew9oFKNsYMvGn66JYShX/ELaJiBwiT935kt3i0fULXxMVIQBeYAnoemXCOksqb5CdZtGy/xjaxdpi5YC78DohI+EMLKayt3ZG4Eo3a5Rig7zbAbgZOO93WbMdG0T8NIxU9s678Jr3J3gXRrR/O+QHg+E6AgIgAAAAAAAAAALhBNdA9clFkeWxecZEjQI3ll9SCf38roqT4KCGg7QmyIJhVBXqLb3j7h1Lsv01TaK0RPzdeKnj1kx/FOqsF3t38pQH4ToABiAAAAAAAAAAAuEENmgkLHGuq0fG0z8dCyCr97o0FZyhjPKH8Fv5fnkpDdHUE6aNswLcGREZjdcfFq304IC7x/LhPTJ0rYq1KKrMIAPhOgAKIAAAAAAAAAAC4QQukLxSNLy4/8RGCSnXKcLyi0R8YfPReTPKAaCoIrNcKb2nF7/1Cs3bYpD+NG4Fn8TuoRNOw+zg06QXA80sGhuwA+E6AA4gAAAAAAAAAALhBm9lwzGylbIUHSrFvbiHit1qVSDI2/2oC2PjlPeRz9ygO/QzSyPeISJhgUwl93r21voZgOjPzqlCYIdETZPxvCwD4ToAEiAAAAAAAAAAAuEFXxF4Q7Xcijaw6XHMBrYEU7FRwU1zktyoyxTlI4J90VmgG1Wx+xSIhAjS7X5l94Y6BndTk541tiyS8zoWjCuXJAfhOgAWIAAAAAAAAAAC4QS+lMnWnDY0OI90ZMJ6mZ2j7j+Wbo/hyAje5+qGrwbZdDSwLNyCz2la22o0Sj5zCBsZxGttVUqtrfOnGT4eZ3SIA","Witness":"hhNJW2B0g6VGfxgky2BUVa5YmbZ+aghlSuxEfjZQNkFnj8QDyNyNeJo3ggY/w9KDbQ8OruSg23HR4X00wUSBYQE=","Type":65,"BlockIndex":33584,"ValidatorIndex":3,"ViewNumber":0}}
2026-03-05T08:20:10.806Z	DEBUG	received message	{"type": "PrepareResponse", "from": 0, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	DEBUG	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": true, "moreThanF": true}
2026-03-05T08:20:10.806Z	DEBUG	received message	{"type": "PrepareResponse", "from": 1, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	DEBUG	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": true, "moreThanF": true}
2026-03-05T08:20:10.806Z	DEBUG	received message	{"type": "PrepareResponse", "from": 2, "height": 33584, "view": 0, "my_height": 33584, "my_view": 0}
2026-03-05T08:20:10.806Z	INFO	received PrepareResponse	{"validator": 2}
2026-03-05T08:20:10.806Z	DEBUG	check preparations	{"hasReq": true, "count": 5, "M": 5}
2026-03-05T08:20:10.806Z	INFO	sending PreCommit	{"height": 33584, "view": 0}
2026-03-05T08:20:10.806Z	DEBUG	broadcasting message	{"type": "PreCommit", "height": 33584, "view": 0}
2026-03-05T08:20:10.806Z	DEBUG	reset timer	{"h": 33584, "v": 0, "delay": "5s"}
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.806Z	ERROR	failed to construct Commit	{"error": "failed to construct Header"}
2026-03-05T08:20:10.806Z	DEBUG	reset timer	{"h": 33584, "v": 0, "delay": "5s"}
WARN [03-05|08:20:10.806] can't construct block due to missing state: proposal verification failed
2026-03-05T08:20:10.807Z	DEBUG	recovering finished cv=0/7 preq=0/0 presp=3/7 pco=0/7 co=0/7
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x10d9462]

goroutine 8505 [running]:
github.com/nspcc-dev/dbft.(*DBFT[...]).checkCommit(0x2630f20)
	github.com/nspcc-dev/dbft@v0.3.2/check.go:128 +0x3c2
github.com/nspcc-dev/dbft.(*DBFT[...]).checkPreCommit(0x2630f20)
	github.com/nspcc-dev/dbft@v0.3.2/check.go:98 +0x9c5
github.com/nspcc-dev/dbft.(*DBFT[...]).checkPrepare(0x2630f20)
	github.com/nspcc-dev/dbft@v0.3.2/check.go:43 +0x63e
github.com/nspcc-dev/dbft.(*DBFT[...]).onPrepareResponse(0x2630f20, {0x260d648, 0xc015172c60})
	github.com/nspcc-dev/dbft@v0.3.2/dbft.go:493 +0xe05
github.com/nspcc-dev/dbft.(*DBFT[...]).OnReceive(0x2630f20, {0x260d648, 0xc015172c60})
	github.com/nspcc-dev/dbft@v0.3.2/dbft.go:271 +0xd98
github.com/nspcc-dev/dbft.(*DBFT[...]).onRecoveryMessage(0x2630f20, {0x260d648, 0xc015172ab0})
	github.com/nspcc-dev/dbft@v0.3.2/dbft.go:693 +0x72d
github.com/nspcc-dev/dbft.(*DBFT[...]).OnReceive(0x2630f20, {0x260d648, 0xc015172ab0})
	github.com/nspcc-dev/dbft@v0.3.2/dbft.go:285 +0x104b
github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).eventLoop(0xc000121188)
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:2355 +0x978
created by github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).Start in goroutine 8479
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:2212 +0x992

````

--- a/consensus/dbft/dbft.go
+++ b/consensus/dbft/dbft.go
@@ -2489,6 +2489,7 @@ func (c *DBFT) OnTransaction(txs []*types.Transaction) {
 	var cbList = c.txCbList.Load()
 	if cbList != nil {
 		for _, tx := range txs {
+			tx = tx.WithoutBlobTxSidecar()
 			_, found := slices.BinarySearchFunc(cbList.([]common.Hash), tx.Hash(), common.Hash.Cmp)
 			if found {
 				c.txs <- tx

